protocols/swap/contracts/Swap.sol

```diff
@@ -219,45 +213,87 @@ contract Swap is ISwap {
   }
 
   /**
-    * @notice Authorize a delegate
+    * @notice Authorize a delegated sender
     * @dev Emits an Authorize event
-    * @param _delegate address
+    * @param _authorizedSender address to authorize
+    * @param _expiry uint256
+    */
+  function authorizeSender(
+    address _authorizedSender,
+    uint256 _expiry
+  ) external {
+    require(msg.sender != _authorizedSender, "INVALID_AUTH_SENDER");
+    require(_expiry > block.timestamp, "INVALID_AUTH_EXPIRY");
+    senderAuthorizations[msg.sender][_authorizedSender] = _expiry;
+    emit AuthorizeSender(msg.sender, _authorizedSender, _expiry);
+  }
+
+  /**
+    * @notice Authorize a delegated signer
+    * @dev Emits an Authorize event
+    * @param _authorizedSigner address to authorize
     * @param _expiry uint256
     */
-  function authorize(
-    address _delegate,
+  function authorizeSigner(
+    address _authorizedSigner,
     uint256 _expiry
   ) external {
-    require(msg.sender != _delegate, "INVALID_AUTH_DELEGATE");
+    require(msg.sender != _authorizedSigner, "INVALID_AUTH_SIGNER");
     require(_expiry > block.timestamp, "INVALID_AUTH_EXPIRY");
-    delegateApprovals[msg.sender][_delegate] = _expiry;
-    emit Authorize(msg.sender, _delegate, _expiry);
+    signerAuthorizations[msg.sender][_authorizedSigner] = _expiry;
+    emit AuthorizeSigner(msg.sender, _authorizedSigner, _expiry);
   }
 
   /**
-    * @notice Revoke an authorization
+    * @notice Revoke an authorized sender
     * @dev Emits a Revoke event
+    * @param _authorizedSender address
+    */
+  function revokeSender(
+    address _authorizedSender
+  ) external {
+    delete senderAuthorizations[msg.sender][_authorizedSender];
+    emit RevokeSender(msg.sender, _authorizedSender);
+  }
+
+  /**
+    * @notice Revoke an authorized signer
+    * @dev Emits a Revoke event
+    * @param _authorizedSigner address
+    */
+  function revokeSigner(
+    address _authorizedSigner
+  ) external {
+    delete signerAuthorizations[msg.sender][_authorizedSigner];
+    emit RevokeSigner(msg.sender, _authorizedSigner);
+  }
+
+  /**
+    * @notice Determine whether a sender delegate is authorized
+    * @param _approver address
     * @param _delegate address
+    * @return bool returns whether a delegate is sender authorized
     */
-  function revoke(
+  function isSenderAuthorized(
+    address _approver,
     address _delegate
-  ) external {
-    delete delegateApprovals[msg.sender][_delegate];
-    emit Revoke(msg.sender, _delegate);
+  ) internal view returns (bool) {
+    if (_approver == _delegate) return true;
```

Comment:
could reduce this line and the next line into 1

Reply:
sweet done!